test/pdevs_partial_accumulator_model_test.cpp

```diff
@@ -0,0 +1,159 @@
+
+            typename cadmium::make_message_bags<floating_partial_accumulator::input_ports>::type bags_two;
+            cadmium::get_messages<typename floating_partial_accumulator_defs::add>(bags_two).push_back(3.0f);
+            cadmium::get_messages<typename floating_partial_accumulator_defs::add>(bags_two).push_back(7.0f);
+            g.external_transition(9.0f, bags_two);
+            //validate state
+            BOOST_CHECK_EQUAL(25.0f, std::get<float>(g.state));
+            BOOST_CHECK_EQUAL(false, std::get<bool>(g.state));
+            BOOST_CHECK_EQUAL(0, std::get<int>(g.state));
+
+            typename cadmium::make_message_bags<floating_partial_accumulator::input_ports>::type bags_three;
+            cadmium::get_messages<typename floating_partial_accumulator_defs::add>(bags_three).push_back(3.0f);
+            cadmium::get_messages<typename floating_partial_accumulator_defs::partial>(bags_three).emplace_back();
+            g.external_transition(2.0f, bags_three);
+            BOOST_CHECK_EQUAL(28.0f, std::get<float>(g.state));
+            BOOST_CHECK_EQUAL(false, std::get<bool>(g.state));
+            BOOST_CHECK_EQUAL(1, std::get<int>(g.state));
+
+            //validate output
+            typename cadmium::make_message_bags<floating_partial_accumulator::output_ports>::type outmb1;
+            outmb1 = g.output();
+            typename cadmium::make_message_bags<floating_partial_accumulator::output_ports>::type outmb_expected;
+            cadmium::get_messages<typename floating_partial_accumulator_defs::sum>(outmb_expected).push_back(28.0f);
+            BOOST_CHECK_EQUAL(1, cadmium::get_messages<typename floating_partial_accumulator_defs::sum>(outmb1).size());
+            BOOST_CHECK_EQUAL(1, cadmium::get_messages<typename floating_partial_accumulator_defs::sum>(outmb_expected).size());
+            BOOST_CHECK_EQUAL(cadmium::get_messages<typename floating_partial_accumulator_defs::sum>(outmb_expected)[0], cadmium::get_messages<typename floating_partial_accumulator_defs::sum>(outmb1)[0]);
+
+            //running confluence, because waiting for an internal here
+            typename cadmium::make_message_bags<floating_partial_accumulator::input_ports>::type bags_four;
+            cadmium::get_messages<typename floating_partial_accumulator_defs::reset>(bags_four).emplace_back();
+            g.confluence_transition(0.0f, bags_four);
+            BOOST_CHECK_EQUAL(28.0f, std::get<float>(g.state));
+            BOOST_CHECK_EQUAL(true, std::get<bool>(g.state));
+            BOOST_CHECK_EQUAL(0, std::get<int>(g.state));
+            
+             //validate output
+            typename cadmium::make_message_bags<floating_partial_accumulator::output_ports>::type outmb2;
+            outmb2 = g.output();
+            BOOST_CHECK_EQUAL(true, cadmium::get_messages<typename floating_partial_accumulator_defs::sum>(outmb2).empty());
+        }
+
+    BOOST_AUTO_TEST_SUITE_END()
+

```

Comment:
Some of them were already done, I have added more combinations. They are inside  BOOST_AUTO_TEST_CASE( output_function_returns_accumulated_value_test_and_external_combinations )